npm_and_yarn/lib/dependabot/npm_and_yarn/helpers.rb

```diff
@@ -255,29 +257,83 @@ def self.run_yarn_commands(*commands)
         commands.each { |cmd, fingerprint| run_single_yarn_command(cmd, fingerprint: fingerprint) }
       end
 
+      # Setup yarn and run a single yarn command returning stdout/stderr
+      sig { params(command: String, fingerprint: T.nilable(String)).returns(String) }
+      def self.run_yarn_command(command, fingerprint: nil)
+        setup_yarn_berry
```

Comment:
we should check to see if we have run the setup command already and skip it if we have

Reply:
also, since most of the codebase uses #run_single_yarn_command, should we update that method instead?

Reply:
Thanks for the feedback. I will check if we can run the `run_single_yarn_command` and make changes accordingly.

Reply:
If you’re suggesting that we configure `setup_yarn_berry` once and avoid re-running these commands, I can work on that. However, I think it would be better suited to a separate PR, as our priority is first ensuring everything runs smoothly with Corepack.